plugins/outputs/zabbix/zabbix.go

```diff
@@ -0,0 +1,199 @@
+package zabbix
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// ZabbixMetric class.
+type ZabbixMetric struct {
+	Host  string `json:"host"`
+	Key   string `json:"key"`
+	Value string `json:"value"`
+	Clock int64  `json:"clock"`
+}
+
+// ZabbixMetric class constructor.
+func NewZabbixMetric(host, key, value string, clock ...int64) *ZabbixMetric {
+	m := &ZabbixMetric{Host: host, Key: key, Value: value}
+	// use current time, if `clock` is not specified
+	if m.Clock = time.Now().Unix(); len(clock) > 0 {
+		m.Clock = int64(clock[0])
+	}
+	return m
+}
+
+// ZabbixPacket class.
+type ZabbixPacket struct {
+	Request string    `json:"request"`
+	Data    []*ZabbixMetric `json:"data"`
+	Clock   int64     `json:"clock"`
+}
+
+// ZabbixPacket class cunstructor.
+func NewZabbixPacket(data []*ZabbixMetric, clock ...int64) *ZabbixPacket {
+	p := &ZabbixPacket{Request: `sender data`, Data: data}
+	// use current time, if `clock` is not specified
+	if p.Clock = time.Now().Unix(); len(clock) > 0 {
+		p.Clock = int64(clock[0])
+	}
+	return p
+}
+
+// DataLen ZabbixPacket class method, return 8 bytes with packet length in little endian order.
+func (p *ZabbixPacket) DataLen() []byte {
+	dataLen := make([]byte, 8)
+	JSONData, _ := json.Marshal(p)
+	binary.LittleEndian.PutUint32(dataLen, uint32(len(JSONData)))
+	return dataLen
+}
+
+type Zabbix struct {
+	Host    string
+	Port    int
+	Hosttag string
+
+	Debug bool
+}
+
+var sampleConfig = `
+	# Address of zabbix host
+	host = "zabbix.example.com"
+	# Port of the Zabbix server
+	port = 10051
+	# Which tag will be used for measurement hostname
+	hosttag = "host"
+`
+
+func (z *Zabbix) Connect() error {
+	// Test connection to Zabbix server
+	// format: hostname:port
+	uri := fmt.Sprintf("%s:%d", z.Host, z.Port)
+	tcpAddr, err := net.ResolveTCPAddr("tcp", uri)
+	if err != nil {
+		return fmt.Errorf("Zabbix: TCP address cannot be resolved")
+	}
+
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
```

Comment:
You should store `connection` on the Zabbix struct so that you don't need to resolve and dial on every write.

Reply:
I have trying to fix that but zabbix closes the connection after the send, so you have to open a new connection for each send.